### ISSUE TITLE
[axe-core warning] - Ensures aria-hidden elements do not contain focusable elements

### ISSUE THREAD

**Issue by bjankord** [cerner/terra-framework#652]
# Feature Request

When the abstract modal is open, we set aria-hidden on the root element to prevent VoiceOver on iOS from shifting the virtual cursor outside of the modal. 

However, we don't currently do anything to disable children within the aria-hidden="true" element.

This PR should fix part of the issue once #608 is merged.
We'll have to add the inert polyfill to terra overlay as it uses similar logic using aria-hidden.

## Description
<details>
 <summary>WDIO Axe Error Report</summary>

```
 Notification-dialog with No Message is accessible:
expected no accessibility violations but got:
        {
  "description": "Ensures aria-hidden elements do not contain focusable elements",
  "help": "ARIA hidden element must not contain focusable elements",
  "helpUrl": "https://dequeuniversity.com/rules/axe/3.2/aria-hidden-focus?application=axeAPI",
  "id": "aria-hidden-focus",
  "impact": "serious",
  "nodes": [
    {
      "all": [
        {
          "data": null,
          "id": "focusable-disabled",
          "impact": "serious",
          "message": "Focusable content should be disabled or be removed from the DOM",
          "relatedNodes": [
            {
              "html": "<button id=\"trigger-notification-dialog\" class=\"Button-module__button___3afIE Button-module__neutral___ZyOh8\" type=\"button\" aria-disabled=\"false\"><span class=\"Button-module__button-label___1dkzN Button-module__text-only___1GgY_\"><span class=\"\">Trigger NotificationDialog</span></span></button>",
              "target": [
                "#trigger-notification-dialog"
              ]
            }
          ]
        }
      ],
      "any": [],
      "failureSummary": "Fix all of the following:\n  Focusable content should be disabled or be removed from the DOM",
      "html": "<div id=\"root\" aria-hidden=\"true\">",
      "impact": "serious",
      "none": [],
      "target": [
        "#root"
      ]
    }
  ],
  "tags": [
    "cat.name-role-value",
    "wcag2a",
    "wcag412"
  ]
}
running chrome
AssertionError: expected no accessibility violations but got:
        {
  "description": "Ensures aria-hidden elements do not contain focusable elements",
  "help": "ARIA hidden element must not contain focusable elements",
  "helpUrl": "https://dequeuniversity.com/rules/axe/3.2/aria-hidden-focus?application=axeAPI",
  "id": "aria-hidden-focus",
  "impact": "serious",
  "nodes": [
    {
      "all": [
        {
          "data": null,
          "id": "focusable-disabled",
          "impact": "serious",
          "message": "Focusable content should be disabled or be removed from the DOM",
          "relatedNodes": [
            {
              "html": "<button id=\"trigger-notification-dialog\" class=\"Button-module__button___3afIE Button-module__neutral___ZyOh8\" type=\"button\" aria-disabled=\"false\"><span class=\"Button-module__button-label___1dkzN Button-module__text-only___1GgY_\"><span class=\"\">Trigger NotificationDialog</span></span></button>",
              "target": [
                "#trigger-notification-dialog"
              ]
            }
          ]
        }
      ],
      "any": [],
      "failureSummary": "Fix all of the following:\n  Focusable content should be disabled or be removed from the DOM",
      "html": "<div id=\"root\" aria-hidden=\"true\">",
      "impact": "serious",
      "none": [],
      "target": [
        "#root"
      ]
    }
  ],
  "tags": [
    "cat.name-role-value",
    "wcag2a",
    "wcag412"
  ]
}
    at Context.<anonymous> (/Users/bj031910/repos/terra-framework/node_modules/terra-toolkit/lib/wdio/services/TerraCommands/accessiblity.js:13:54)
    at new Promise (<anonymous>)
    at new F (/Users/bj031910/repos/terra-framework/node_modules/core-js/library/modules/_export.js:36:28)

```
</details>

<details>
 <summary>WDIO Axe Error Report</summary>
```
1) ApplicationLayout Toggles menu when small is accessible:
expected no accessibility violations but got:
        {
  "description": "Ensures aria-hidden elements do not contain focusable elements",
  "help": "ARIA hidden element must not contain focusable elements",
  "helpUrl": "https://dequeuniversity.com/rules/axe/3.2/aria-hidden-focus?application=axeAPI",
  "id": "aria-hidden-focus",
  "impact": "serious",
  "nodes": [
    {
      "all": [
        {
          "data": null,
          "id": "focusable-disabled",
          "impact": "serious",
          "message": "Focusable content should be disabled or be removed from the DOM",
          "relatedNodes": [
            {
              "html": "<button type=\"button\" class=\"ApplicationHeader-module__toggle-button___2GANm\" aria-label=\"Toggle Menu\" data-application-header-toggle=\"true\">",
              "target": [
                ".ApplicationHeader-module__toggle-button___2GANm"
              ]
            },
            {
              "html": "<button type=\"button\" class=\"ApplicationHeaderLayout-module__skip-content___1vGmQ\">Skip to Content</button>",
              "target": [
                ".ApplicationHeaderLayout-module__skip-content___1vGmQ"
              ]
            }
          ]
        }
      ],
      "any": [],
      "failureSummary": "Fix all of the following:\n  Focusable content should be disabled or be removed from the DOM",
      "html": "<main data-terra-layout-main=\"true\" tabindex=\"-1\" class=\"LayoutSlidePanel-module__main-container___1O3uK\" aria-hidden=\"true\">",
      "impact": "serious",
      "none": [],
      "target": [
        "main"
      ]
    }
  ],
  "tags": [
    "cat.name-role-value",
    "wcag2a",
    "wcag412"
  ]
}
running chrome
AssertionError: expected no accessibility violations but got:
        {
  "description": "Ensures aria-hidden elements do not contain focusable elements",
  "help": "ARIA hidden element must not contain focusable elements",
  "helpUrl": "https://dequeuniversity.com/rules/axe/3.2/aria-hidden-focus?application=axeAPI",
  "id": "aria-hidden-focus",
  "impact": "serious",
  "nodes": [
    {
      "all": [
        {
          "data": null,
          "id": "focusable-disabled",
          "impact": "serious",
          "message": "Focusable content should be disabled or be removed from the DOM",
          "relatedNodes": [
            {
              "html": "<button type=\"button\" class=\"ApplicationHeader-module__toggle-button___2GANm\" aria-label=\"Toggle Menu\" data-application-header-toggle=\"true\">",
              "target": [
                ".ApplicationHeader-module__toggle-button___2GANm"
              ]
            },
            {
              "html": "<button type=\"button\" class=\"ApplicationHeaderLayout-module__skip-content___1vGmQ\">Skip to Content</button>",
              "target": [
                ".ApplicationHeaderLayout-module__skip-content___1vGmQ"
              ]
            }
          ]
        }
      ],
      "any": [],
      "failureSummary": "Fix all of the following:\n  Focusable content should be disabled or be removed from the DOM",
      "html": "<main data-terra-layout-main=\"true\" tabindex=\"-1\" class=\"LayoutSlidePanel-module__main-container___1O3uK\" aria-hidden=\"true\">",
      "impact": "serious",
      "none": [],
      "target": [
        "main"
      ]
    }
  ],
  "tags": [
    "cat.name-role-value",
    "wcag2a",
    "wcag412"
  ]
}
    at Context.<anonymous> (/Users/bj031910/repos/terra-framework/node_modules/terra-toolkit/lib/wdio/services/TerraCommands/accessiblity.js:13:54)
    at new Promise (<anonymous>)
    at new F (/Users/bj031910/repos/terra-framework/node_modules/core-js/library/modules/_export.js:36:28)

2) ApplicationLayout Renders primary nav menu when small is accessible:
expected no accessibility violations but got:
        {
  "description": "Ensures aria-hidden elements do not contain focusable elements",
  "help": "ARIA hidden element must not contain focusable elements",
  "helpUrl": "https://dequeuniversity.com/rules/axe/3.2/aria-hidden-focus?application=axeAPI",
  "id": "aria-hidden-focus",
  "impact": "serious",
  "nodes": [
    {
      "all": [
        {
          "data": null,
          "id": "focusable-disabled",
          "impact": "serious",
          "message": "Focusable content should be disabled or be removed from the DOM",
          "relatedNodes": [
            {
              "html": "<button type=\"button\" class=\"ApplicationHeader-module__toggle-button___2GANm\" aria-label=\"Toggle Menu\" data-application-header-toggle=\"true\">",
              "target": [
                ".ApplicationHeader-module__toggle-button___2GANm"
              ]
            },
            {
              "html": "<button type=\"button\" class=\"ApplicationHeaderLayout-module__skip-content___1vGmQ\">Skip to Content</button>",
              "target": [
                ".ApplicationHeaderLayout-module__skip-content___1vGmQ"
              ]
            }
          ]
        }
      ],
      "any": [],
      "failureSummary": "Fix all of the following:\n  Focusable content should be disabled or be removed from the DOM",
      "html": "<main data-terra-layout-main=\"true\" tabindex=\"-1\" class=\"LayoutSlidePanel-module__main-container___1O3uK\" aria-hidden=\"true\">",
      "impact": "serious",
      "none": [],
      "target": [
        "main"
      ]
    }
  ],
  "tags": [
    "cat.name-role-value",
    "wcag2a",
    "wcag412"
  ]
}
running chrome
AssertionError: expected no accessibility violations but got:
        {
  "description": "Ensures aria-hidden elements do not contain focusable elements",
  "help": "ARIA hidden element must not contain focusable elements",
  "helpUrl": "https://dequeuniversity.com/rules/axe/3.2/aria-hidden-focus?application=axeAPI",
  "id": "aria-hidden-focus",
  "impact": "serious",
  "nodes": [
    {
      "all": [
        {
          "data": null,
          "id": "focusable-disabled",
          "impact": "serious",
          "message": "Focusable content should be disabled or be removed from the DOM",
          "relatedNodes": [
            {
              "html": "<button type=\"button\" class=\"ApplicationHeader-module__toggle-button___2GANm\" aria-label=\"Toggle Menu\" data-application-header-toggle=\"true\">",
              "target": [
                ".ApplicationHeader-module__toggle-button___2GANm"
              ]
            },
            {
              "html": "<button type=\"button\" class=\"ApplicationHeaderLayout-module__skip-content___1vGmQ\">Skip to Content</button>",
              "target": [
                ".ApplicationHeaderLayout-module__skip-content___1vGmQ"
              ]
            }
          ]
        }
      ],
      "any": [],
      "failureSummary": "Fix all of the following:\n  Focusable content should be disabled or be removed from the DOM",
      "html": "<main data-terra-layout-main=\"true\" tabindex=\"-1\" class=\"LayoutSlidePanel-module__main-container___1O3uK\" aria-hidden=\"true\">",
      "impact": "serious",
      "none": [],
      "target": [
        "main"
      ]
    }
  ],
  "tags": [
    "cat.name-role-value",
    "wcag2a",
    "wcag412"
  ]
}
    at Context.<anonymous> (/Users/bj031910/repos/terra-framework/node_modules/terra-toolkit/lib/wdio/services/TerraCommands/accessiblity.js:13:54)
    at new Promise (<anonymous>)
    at new F (/Users/bj031910/repos/terra-framework/node_modules/core-js/library/modules/_export.js:36:28)
```
</details>

**Comment by bjankord** [comment]
This has been resolved in [a prior PR](https://github.com/cerner/terra-framework/pull/608) and is no longer being reported when running axe-core with the WCAG 2.1 AA ruleset.
